### PR TITLE
tycho2: initial commit

### PR DESCRIPTION
**Pull request by junghans**
/cc @ckrisgarrett

### CODE REVIEW

File: var/spack/repos/builtin/packages/tycho2/package.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    depends_on("mpi")
+
+    def patch(self):
+        shutil.move('make.inc.example', 'make.inc')
+
+    @property
+    def build_targets(self):
+        targets = [
+            'MPICC={0} -std=c++11 -fopenmp'.format(self.spec['mpi'].mpicxx)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/tycho2/package.py]: This OpenMP flag wrong work for every compiler. Please use `self.compiler.openmp_flag`.

File: var/spack/repos/builtin/packages/tycho2/package.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        ]
+
+        return targets
+
+    def install(self, spec, prefix):
+        mkdirp(prefix.bin)
+        for exe in ["sweep.x"]:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/tycho2/package.py]: Why a for loop for a single element list?

File: var/spack/repos/builtin/packages/tycho2/package.py (hunk 1 above)
Comment (junghans) [inline var/spack/repos/builtin/packages/tycho2/package.py]: Good point!

Comment (junghans) [inline var/spack/repos/builtin/packages/tycho2/package.py]: @adamjstewart: BTW, is there a way to check if the compiler support OpenMP?

**Comment by junghans** [comment]
ping @adamjstewart

**Comment by junghans** [comment]
@scheibelp

File: var/spack/repos/builtin/packages/tycho2/package.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    """
+    homepage = "https://github.com/lanl/tycho2"
+    url      = "https://github.com/lanl/tycho2/tarball/v0.1"
+
+    version('develop', git='https://github.com/lanl/tycho2', branch='master')
+
+    depends_on("mpi")
+
+    def patch(self):
+        touch('make.inc')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/tycho2/package.py]: Could you add an explanation of why this is needed in a comment? Sorry if that's obvious but I don't see any examples in other spack packages.

Comment (junghans) [reply var/spack/repos/builtin/packages/tycho2/package.py]: Good point!

File: var/spack/repos/builtin/packages/tycho2/package.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    """
+    homepage = "https://github.com/lanl/tycho2"
+    url      = "https://github.com/lanl/tycho2/tarball/v0.1"
+
+    version('develop', git='https://github.com/lanl/tycho2', branch='master')
+
+    depends_on("mpi")
+
+    def patch(self):
+        touch('make.inc')  # included by Makefile, but we set MPICC below
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/tycho2/package.py]: I appreciate the edit but I'm still not clear on why this is required. Does setting MPICC result in this not being generated when it normally would be?

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**